### PR TITLE
Increase preStop delay for 35 seconds to allow DNS TTL to expire and die gracefully without spilling over requests

### PR DESCRIPTION
**Pull request by uthark**
### References
- Jira link: PAAS-3874

### Risks
- Low

### CODE REVIEW

**Comment by grosser** [comment]
- we should make this an ENV var so we don't hardcode values that work for us
 - we should set terminationgraceperiod to time + x so we don't sigkill the processes ... ideally set to "[user-configured, 35+x].max" ... but that get's fishy too ... making the DNS cache be <30s would be ideal so we don't have to think about terminationgraceperiod

**Comment by uthark** [comment]
I've added environment variable. Reducing DNS TTL will help, but setting grace period doesn't harm.

**Review by grosser — APPROVED** [review]

File: plugins/kubernetes/app/models/kubernetes/template_filler.rb [hunk 1]
```diff
@@ -569,11 +570,20 @@ def set_image_pull_secrets
...
       return unless KUBERNETES_ADD_PRESTOP
       pod_containers.each do |container|
         next if samson_container_config(container, :"samson/preStop") == "disabled"
-        (container[:lifecycle] ||= {})[:preStop] ||= {exec: {command: ["sleep", "3"]}}
+        # set sleep to 35 seconds to allow for DNS TTL to expire.
+        # Default termination grace period for pods
+        (container[:lifecycle] ||= {})[:preStop] ||= {exec: {command: ["sleep", pre_stop_sleep.to_s]}}
+      end
+      if pre_stop_sleep.to_i > DEFAULT_TERMINATION_GRACE_PERIOD
+        pod_template[:spec][:terminationGracePeriodSeconds] = pre_stop_sleep+3 # add 3 extra seconds before terminating.
```
Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/template_filler.rb]:
```suggestion
        pod_template[:spec][:terminationGracePeriodSeconds] = pre_stop_sleep + 3 # make sure servers shut down after they are done waiting
```

File: plugins/kubernetes/test/models/kubernetes/template_filler_test.rb [hunk 2]
```diff
@@ -850,6 +850,12 @@ def secret_annotations(hash)
           )
         end
 
+        it "sets termination grace period if sleep time > default" do
```
Comment (grosser) [inline plugins/kubernetes/test/models/kubernetes/template_filler_test.rb]:
```suggestion
        it "bumps termination grace period if we would sleep longer that termination allows" do
```

**Pull request merged**